src/components/Header.js

```diff
@@ -1,7 +1,24 @@
-import React from "react";
+import React, { Fragment, useState } from "react";
+import SettingsModal from "./SettingsModal";
 
 const Header = () => {
-  return <header className="header">What's near me?</header>;
+  const [displayModal, setDisplayModal] = useState(false);
+
+  return (
+    <Fragment>
+      {displayModal && <SettingsModal setDisplayModal={setDisplayModal} />}
+
+      <header className="header">
+        What's near me?
+        <button
+          className="header-settings-btn"
+          onClick={() => setDisplayModal(true)}
+        >
+          <img src="/img/settings.png" alt="gear icon" />
+        </button>
+      </header>
+    </Fragment>
+  );
```

Comment:
excellent, this is how we worked with Modals with the shopping list task very simple and to the point